core/src/main/scala/org/http4s/Header.scala

```diff
@@ -132,7 +132,16 @@ object Header {
     _.toString
   }
 
-  implicit val HeaderEq: Eq[Header] = Eq.instance[Header] { (a, b) =>
-    a.name === b.name && a.value === b.value
-  }
+  @deprecated(message = "Please use HeaderOrder instead", since = "1.0.0-M8")
+  lazy val HeaderEq: Eq[Header] = HeaderOrder
```

Comment:
Nit: since it's just an alias, a `def` is faster and lighter.

Reply:
I'll update this.